src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs

```diff
@@ -30,13 +32,63 @@ public partial class GraphToolbox : UserControl
 
     private object? _deleteCommandParameter;
 
+    public static readonly DirectProperty<GraphToolbox, string?> TextProperty =
+        AvaloniaProperty.RegisterDirect<GraphToolbox, string?>(
+            nameof(Text),
+            o => o.Text,
+            (o, v) => o.Text = v);
+
+    public string? Text
+    {
+        get => _text;
+        set => SetAndRaise(TextProperty, ref _text, value);
+    }
+
+    private string? _text;
+
+    private readonly TextBox _renameBox;
+    private readonly TextBlock _nameBlock;
+
     public GraphToolbox()
     {
         InitializeComponent();
+
+        _renameBox = this.FindControl<TextBox>("RenameBox");
+        _nameBlock = this.FindControl<TextBlock>("NameBlock");
     }
 
     private void InitializeComponent()
     {
         AvaloniaXamlLoader.Load(this);
     }
+
+    private void RenameButton_OnClick(object? sender, RoutedEventArgs e)
+    {
+        SetRenameMode(true);
+    }
+
+    private void RenameBox_OnKeyUp(object? sender, KeyEventArgs e)
+    {
+        if (e.Key == Key.Enter)
+        {
+           SetRenameMode(false);
+        }
+    }
```

Comment:
Idea for a future PR -- on `Escape` cancel changes and keep the original name of the graph.

Reply:
Also done :)